firmware/console/binary/bluetooth.cpp

```diff
@@ -108,11 +108,11 @@ static int btVerOk(SerialTsChannelBase* tsChannel)
 {
 	int len;
 	int ret = -1;
-	char tmp[38];
+	char tmp[46];
 
 	/* wait for resposne  */
 	len = btReadLine(tsChannel, tmp, sizeof(tmp));
-	if ((len >= 36) && (len <= 40)) {
+	if ((len >= 36) && (len <= 46)) {
```

Comment:
Hi Jcor25, what did you discover here?

Reply:
this is for JDY-31 only. jdy31 do not respond to AT or AT+DISCOVERY, only AT+VERSION, update is for the newer batches that have longer response. now that I'm thinking about it you can also check for a known response such as AT+BAUD, response wound be AT+BAUD(?) so same amount of characters.